src/coreclr/vm/peimagelayout.cpp

```diff
@@ -721,14 +729,57 @@ FlatImageLayout::FlatImageLayout(PEImage* pOwner)
         _ASSERTE((offset - mapBegin) < mapSize);
         _ASSERTE(mapSize >= (UINT64)size);
 
-        char *addr = (char*)CLRMapViewOfFile(m_FileMap, FILE_MAP_READ, mapBegin >> 32, (DWORD)mapBegin, (DWORD)mapSize);
-        if (addr == NULL)
+        LPVOID view = CLRMapViewOfFile(m_FileMap, FILE_MAP_READ, mapBegin >> 32, (DWORD)mapBegin, (DWORD)mapSize);
+        if (view == NULL)
             ThrowLastError();
 
-        addr += (offset - mapBegin);
-        m_FileView.Assign((LPVOID)addr);
+        m_FileView.Assign(view);
+        addr = (LPVOID)((size_t)view + offset - mapBegin);
```

Comment:
This changes what value gets assigned to `m_FileView`. Is this intentional?

Reply:
Docs say releasing should be done with what `CLRMapViewOfFile` returned. Not sure how it worked before. Perhaps the  view/map was never released.

Reply:
Yes, the change is intentional.

Reply:
Perhaps release rounds to allocation granularity internally (1+ pages) so it still worked. We should keep the original value though.

Reply:
Great - makes sense.